torch/_dynamo/variables/tensor.py

```diff
@@ -100,6 +101,7 @@ def __init__(
         self.is_sparse = is_sparse
         self.class_type = class_type
         self.specialized_value = specialized_value
+        self.tensor_dict = tensor_dict
```

Comment:
There is nothing tensor specific about this handling. All defined in Python objects have `__dict__` and we should uniformly apply this handling to all objects